File: util/Plotter/plot_sweep.py
"""Utilities for plotting parameter sweeps (throughput vs. param)."""
import matplotlib.pyplot as plt
import torch


def plot_sweep(x_values, y_values, xlabel='param', ylabel='value', title=None, show=True, save_path=None):
    plt.figure()
    plt.plot(x_values, y_values, marker='o')
    plt.xlabel(xlabel)
    plt.ylabel(ylabel)
    if title:
        plt.title(title)
    if save_path:
        plt.savefig(save_path)
    if show:
        plt.show()

File: util/Plotter/plot_sweep.m
function plot_sweep (x, sumrate_kmeans_ref_arr, sumrate_kmeans_opt_arr, sumrate_hier_ref_arr, sumrate_hier_opt_arr, xlabelText, figTitle)
figure;
hold on;
plot(x, sumrate_kmeans_ref_arr, '-o', 'DisplayName', 'K-Means Reference');
plot(x, sumrate_kmeans_opt_arr, '-x', 'DisplayName', 'K-Means Optimized');
plot(x, sumrate_hier_ref_arr, '-s', 'DisplayName', 'Hierarchical Reference');
plot(x, sumrate_hier_opt_arr, '-d', 'DisplayName', 'Hierarchical Optimized');
hold off;
xlabel(xlabelText);
ylabel('Sum Rate (bps)');
title(figTitle);
legend show;
grid on;
end